_posts/2016-08-11-React-Native-at-Artsy.markdown

```diff
@@ -0,0 +1,173 @@
+  “Write once, run anywhere.” slogan of yesteryear. The difference being that you can use the same paradigms to create
+  products on various platforms, rather than pure code-reuse.
+
+  While we haven’t put this to the test yet at Artsy, we do plan to team up people across platforms to implement single
+  features on each respective platform, rather than having multiple people implement the same feature _on their own_.
+  The hope is that this will lead to better understanding of features and thus the implementations thereof, while still
+  taking each platform’s unique nature into account. The added benefit would be that people learn to understand and
+  appreciate those unique platform traits, thus making them more well-rounded engineers.
+
+  In the long run, we hope to extend this way of working as we start work on a React Native Android client.
+
+* While Auto Layout is a great step up from manual frame calculation, most of our views don’t need the granularity that
+  Auto Layout offers. React Native ditches Auto Layout and instead uses
+  [flexbox](https://en.wikipedia.org/wiki/CSS_Flex_Box_Layout) for its layout. While my head has never been able to
+  fully wrap around classic CSS, I find that flexbox is an abstraction that nicely fits most of our needs.
+
+  I can definitely imagine situations in which more granularity would be required, however, in those cases we can always
+  decide to ‘drop down’ to native view code, so I don’t really worry too much about that.
+
+* Layout calculations are performed on a background thread, the so-called ‘shadow’ thread. This can make a big
+  difference when e.g. scrolling through a large complex list view.
+
+  Granted, you _can_ do this with `UIView`, but the pattern is not as ingrained in UIKit thus usually leading to more holistic
+  replacements such as [AsyncDisplayKit](http://asyncdisplaykit.org).
+
+* Great separation of declarative view layout (JS, single-threaded) and technical details (native code, multi-threaded).
+  Because of the hard constraint of having a JS/native bridge, there really is no way to take shortcuts (e.g. spaghetti
+  code) that in the long run would only lead to technical debt. Constraints are great.
+
+* Because there’s very little code that needs to be compiled and how the isolated component nature of React makes it
+  it very easy to reload code in-process, development velocity lies much higher than with your typical native UIKit-based
+  development.
+
+* While, as mentioned above, there are definitely issues with the tooling and libs, on the flip-side it is all
+  open-source software and you _can_ (officially) dive in and figure it out, unlike e.g. Xcode and UIKit. (Granted, you
+  still need to deal with these when using React Native, but it can be kept to a minimal surface.)
+
+* Because React Native is still young, fast moving, and open-source, this is a great time to help shape the framework
```

Comment:
It would be interesting to call out where React Native is on the hype cycle.

![image](https://cloud.githubusercontent.com/assets/542335/17572060/ce87ace0-5f20-11e6-8fee-c0a71cdef1fb.png)